botorch/optim/initializers.py

```diff
@@ -274,6 +275,9 @@ def gen_batch_initial_conditions(
         equality constraints: A list of tuples (indices, coefficients, rhs),
             with each tuple encoding an inequality constraint of the form
             `\sum_i (X[indices[i]] * coefficients[i]) = rhs`.
+        generator: Callable for generating samples that are then further
+            processed. It receives `n`, `q` and `seed` as arguments and  returns
```

Comment:
should we make `seed` optional here?

Reply:
I would keep it mandatory, as a seed is always used in `gen_batch_initial_conditions` and the seed number is increased in every iteration.